src/python/examples/unittests/test_data_dumps.py

```diff
@@ -0,0 +1,193 @@
+
+        """
+        type(self).core_dbs = multi_dbs
+        server_url = sqlalchemy.engine.url.make_url(request.config.getoption('server'))
+        type(self).host = server_url.host
+        type(self).port = server_url.port
+        type(self).username = "ensro" if server_url.username == "ensadmin" else server_url.username
+
+    @pytest.mark.skipif(os.environ['USER'] == 'travis',
+                        reason="The test requires both Perl and Python which is not supported by Travis.")
+    @pytest.mark.parametrize(
+        "core_list, species_set_name, id_type, expectation",
+        [
+            (
+                [f"{os.environ['USER']}_gallus_gallus_core_99_6",
+                 f"{os.environ['USER']}_homo_sapiens_core_99_38"],
+                "vertebrates", "protein", does_not_raise()
+            ),
+            ([], "test", "gene", pytest.raises(ValueError))
+        ]
+    )
+    def test_dump_genomes(self, core_list: List[str], species_set_name: str,
+                          tmp_dir: Path, id_type: str, expectation: ContextManager) -> None:
+        """Tests :func:`data_dumps.dump_genomes()` when server connection can be established.
+
+        Args:
+            species_list: A list of species (genome names).
+            species_set_name: Species set (collection) name.
+            tmp_dir: Unit test temp directory (fixture).
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            data_dumps.dump_genomes(core_list, species_set_name, self.host, self.port, tmp_dir,
+                                             id_type)
+
+            out_files = tmp_dir / species_set_name
+            # pylint: disable-next=no-member
+            exp_out = pytest.files_dir / "dump_genomes"  # type: ignore[attr-defined]
+            for db_name, unittest_db in self.core_dbs.items():
+                assert file_cmp(out_files / f"{unittest_db.dbc.db_name}.fasta", exp_out / f"{db_name}.fasta")
+
+    def test_dump_genomes_fake_connection(self, tmp_dir: Path) -> None:
+        """Tests :func:`data_dumps.dump_genomes()` with fake server details.
+
+        Args:
+            tmp_dir: Unit test temp directory (fixture).
+
+        """
+        with pytest.raises(RuntimeError):
+            data_dumps.dump_genomes(["mus_musculus", "naja_naja"], "fake",
+                                             "fake-host", 65536, tmp_dir, "protein")
+
+    def test_dump_genomes_fake_output_path(self) -> None:
+        """Tests :func:`data_dumps.dump_genomes()` with fake output path."""
+        with pytest.raises(OSError):
+            data_dumps.dump_genomes(["mus_musculus", "naja_naja"], "default",
+                                             self.host, self.port, "/compara", "protein")
```

Comment:
I'd be wary of assuming that a path does not exist in the root directory, but if it must be done, I'd suggest at least renaming it to make its purpose clear: 
```suggestion
                                             self.host, self.port, "/nonexistent/path", "protein")
```